lib/contentProfileListener.js

```diff
@@ -89,21 +108,33 @@ class ContentProfileImporter extends SHRContentProfileParserListener {
       // Create new rule, apply it in enterHeaderFlag function
       this._currentRule = new ContentProfileRule();
     }
+    else if (!ctx.headerFlags()) {
+      const pathStr = ctx.contentHeader().simpleName().getText();
+      const identifier = new Identifier(this._currentNs, pathStr);
+      this._currentDef = new ContentProfile(identifier);
+      this._currentDef.grammarVersion = this._currentGrammarVersion;
+      this._currentRule = new ContentProfileRule();
+      this._currentRule.primaryProfile = true;
+      this._currentDef.addRule(this._currentRule);
+    }
   }
 
   enterHeaderFlag(ctx) {
     if (this._currentRule) {
       this._currentRule.noProfile = (ctx.KW_NO_PROFILE() != null);
+      if (this._currentRule.noProfile) this._currentRule.primaryProfile = false;
       this._currentDef.addRule(this._currentRule);
     }
   }
 
   enterContentHeader(ctx) {
     // set current content
-    const name = ctx.simpleName().getText();
-    const identifier = new Identifier(this._currentNs, name);
-    this._currentDef = new ContentProfile(identifier);
-    this._currentDef.grammarVersion = this._currentGrammarVersion;
+    if (this._currentDef == null) {
+      const name = ctx.simpleName().getText();
+      const identifier = new Identifier(this._currentNs, name);
+      this._currentDef = new ContentProfile(identifier);
+      this._currentDef.grammarVersion = this._currentGrammarVersion;
+    }
```

Comment:
I'm not sure that the block above is even necessary anymore.  `enterContentDef` will always be called before `enterContentHeader` -- and as best I can tell, `enterContentDef` should always set `this._currentDef`-- so I doubt the code above ever gets invoked.

Reply:
Correct that block does nothing. It was a vestige from how this used to be implemented that I missed. Removed in https://github.com/standardhealth/shr-text-import/pull/106/commits/27603300c7219801657539ca83a166ce7217b755.